src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -1523,8 +1523,25 @@ private object ReadEnumValue(bool insideJsonObjectValue, IEdmEnumTypeReference e
                 throw new ODataException(ODataErrorStrings.JsonReaderExtensions_UnexpectedNodeDetectedWithPropertyName(JsonNodeType.PrimitiveValue, JsonNodeType.StartObject, propertyName));
             }
 
-            string enumStr = this.JsonReader.ReadStringValue();
-            return new ODataEnumValue(enumStr, expectedValueTypeReference.FullName());
+            object enumValue = this.JsonReader.ReadPrimitiveValue();
+            string enumString;
+            if (enumValue is int enumIntValue)
+            {
+                IEdmEnumType edmEnumType = expectedValueTypeReference.EnumDefinition();
+                IEdmEnumMember selectedMember = edmEnumType.Members.FirstOrDefault(member => member.Value.Value == enumIntValue);
+                enumString = selectedMember.Name;

```

Comment:
Can't `selectedMember` be `null` here because of `FirstOrDefault` potentially not finding an element?